File: backend/src/main/java/com/blog/user/DTO/UserDTO.java
package com.blog.user.DTO;

import java.time.LocalDateTime;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

public class UserDTO {
    @Data
    @AllArgsConstructor
    public static class SearchedUsers {
        private Long id;
        private String userName;
        private String job;
        private String avatar;
        private Boolean followed;
    }

    @Data
    @AllArgsConstructor
    @Builder
    public static class ProfileOutput {
        private Long id;
        private String userName;
        private String job;
        private String avatar;
        private LocalDateTime createdAt;
        private Long followers;
        private Long following;
        private Boolean followed;
        private Boolean MyAccount;
    }

    @Data
    @AllArgsConstructor
    @Builder
    public static class StatiqueInfo {
        private Long usersCount;
        private Long blogsCount;
        private Long reportsCount;
    }

    @Data
    @AllArgsConstructor
    @Builder
    public static class StatiqueUsers {
        private Long usersCount;
        private Long BannnedCount;
        private Long ActiveCount;
    }

    @Data
    @AllArgsConstructor
    @Builder
    public static class UsersData {
        private Long id;
        private String avatar;
        private String role;
        private LocalDateTime joined;
    }
}

File: frontend/src/app/core/shared/interfaces/userDTO.ts
export interface User {
  id: number;
  avatar: string;
  userName: string;
  email: string;
  job: string;
  createdAt: Date;
  role: string;
  token: string;
}


export interface followResponse {
  userId: number;
  status: boolean;
  action: string;
}

export interface ProfileData {
  id: number;
  userName: string;
  job: string;
  avatar: string;
  createdAt: Date;
  followers: number;
  following: number;
  followed: boolean;
  MyAccount: boolean;
}
export interface SearchedUsers {
  id: number;
  userName: string;
  job: string;
  avatar: string;
  followed: boolean;
}


export interface Resolve {
  reportID: number;
  message: string;
}

export interface ActionResponse {
  id: number;
  status: boolean;
}

export interface UsersData {
  id: number;
  avatar: string | null;
  userName: string;
  role: string;
  banned: boolean;
  joined: Date;
}

export interface NotificationsData {
  id: number;
  createdAt: Date;
  readed: boolean;
  userName: string;
  avatar: string | null;
}
